enclave_build/src/docker.rs

```diff
@@ -38,20 +41,25 @@ pub struct DockerUtil {
 
 impl DockerUtil {
     /// Constructor that takes as argument a tag for the docker image to be used
-    pub fn new(docker_image: String) -> Self {
+    pub fn new(docker_image: String) -> Result<Self, DockerError> {
         let mut docker_image = docker_image;
 
         if !docker_image.contains(':') {
             docker_image.push_str(":latest");
         }
 
-        DockerUtil {
-            // DOCKER_HOST environment variable is parsed inside
-            // if docker daemon address needs to be substituted.
-            // By default it tries to connect to 'unix:///var/run/docker.sock'
-            docker: Docker::new(),
+        // DOCKER_HOST environment variable is parsed inside
+        // if docker daemon address needs to be substituted.
+        // By default, it tries to connect to 'unix:///var/run/docker.sock'
+        let docker = Docker::connect_with_defaults().map_err(|e| {
+            error!("{:?}", e);
```

Comment:
nit: Should be added to the `ConnectionError` itself and printed when it is handled